command/bitmarkd/commands.go

```diff
@@ -352,9 +368,19 @@ func dnsTXT(options *Configuration) {
 
 	peering := options.Peering
 
-	publicKey, err := zmqutil.ReadPublicKey(peering.PublicKey)
-	if nil != err {
-		exitwithstatus.Message("error: cannot read public key: %q  error: %s", peering.PublicKey, err)
+	privateKeyBytes, err := hex.DecodeString(peering.PrivateKey)
+	if err != nil {
+		exitwithstatus.Message("error: cannot decode private key: %q  error: %s", peering.PrivateKey, err)
+	}
+
+	privateKey, err := crypto.UnmarshalPrivateKey(privateKeyBytes)
+	if err != nil {
+		exitwithstatus.Message("error: cannot generate private key: %q  error: %s", peering.PrivateKey, err)
+	}
+
+	peerID, err := peer.IDFromPublicKey(privateKey.GetPublic())
```

Comment:
peer.IDFromPrivateKey(privateKey) may be more direct

Reply:
Will fix.